gpumon/gpumon_server.ml

```diff
@@ -23,17 +23,40 @@ module Make(I: Interface) = struct
 
   module Nvidia = struct
     
+    let host_driver_supporting_migration = 385
+    (** Smallest major version of the host driver that supports migration *)
+
     let get_interface_exn () = 
     match I.interface with
     | Some interface -> interface
     | None -> raise Gpumon_interface.NvmlInterfaceNotAvailable
 
     let get_pgpu_metadata _ dbg pgpu_address =
+      let this = "get_pgpu_metadata" in
       let interface = get_interface_exn () in
       try
-        let device = Nvml.device_get_handle_by_pci_bus_id interface pgpu_address in
-        Nvml.device_get_pgpu_metadata interface device
-      with err -> raise (Gpumon_interface.NvmlFailure (Printexc.to_string err))
+        let device =
+          Nvml.device_get_handle_by_pci_bus_id interface pgpu_address in
+        let compat =
+          Nvml.device_get_pgpu_metadata interface device in
+        let version, revision, driver =
+          ( Nvml.pgpu_metadata_get_pgpu_version compat
+          , Nvml.pgpu_metadata_get_pgpu_revision compat
+          , Nvml.pgpu_metadata_get_pgpu_host_driver_version compat
+          ) in
+        let major = Scanf.sscanf driver "%d." (fun x -> x) in
```

Comment:
Should we separate the `Scanf` part from the `Nvml` part in two different `try..with` blocks?

Reply:
That would be to single out failures related to the version format?

Reply:
I agree, probably not worth it

Reply:
This would not work because `compat` is not available for the `if` statement.

Reply:
True